src/components/tabs/TabContent.svelte

```diff
@@ -35,14 +37,63 @@
       const props = tab.content.getProps?.(context) ?? {};
       const tabComponentContext =
         tab.content.getContext?.(allContexts) ?? allContexts;
-      const svelteTabComponent = mount(tab.content.component, {
-        target: tidyTab.value,
-        context: tabComponentContext,
-        props: props,
-      });
+
+      const component =
+        typeof tab.content.component === 'function'
+          ? tab.content.component
+          : tab.content.component.component;
+      // We allow the user to provide their own mount function from their own svelte runtime
+      const { mount: _mount, unmount: _unmount } =
+        'mount' in tab.content.component
+          ? tab.content.component
+          : { mount, unmount };
+
+      const isExternalMount = mount !== _mount;
+      // Both need to be external or internal, else we got problems
+      const isValidMountPair = isExternalMount === (unmount !== _unmount);
+
+      if (!isValidMountPair)
+        throw new Error('Both mount and unmount must be provided if one is');
+
+      let svelteTabComponent: {};
+      if (!isExternalMount) {
+        svelteTabComponent = _mount(component, {
+          target: tidyTab.value,
+          context: tabComponentContext,
+          props: props,
+        });
+      } else {
+        // If we're using their mount, we need to do change how we handle the context so that they can hook into our reactivity
+        const safeTabComponentContext = new Map(
+          tabComponentContext.entries().map(([key, val]) => {
```

Comment:
I'm assume I'm allowed to use this, and that this is polyfilled? Safari only provided this this year, everything else has allowed it in modern browsers as of last year.